### PR TITLE
[release-4.5] Bug 1880333: bump k8s.io/client-go to 0.18.6

### PR DESCRIPTION
**Pull request by kikisdeliveryservice**
- bump k8s.io/client-go to 0.18.16 to pick up fix along with other k8s.io components
- bump kubernets to v1.18.6 to keep version parity.

### CODE REVIEW

**Comment by kikisdeliveryservice** [comment]
@p0lyn0mial PTAL and let me know if you need anything else.

Thanks!

**Comment by kikisdeliveryservice** [comment]
I think I finally got this...

@p0lyn0mial PTAL

**Comment by p0lyn0mial** [comment]
@kikisdeliveryservice I don't see any files in the `vendor` directory. Only changes to the `go.mod` file. Have you forgotten to push some files :)?

**Comment by sinnykumari** [comment]
It needs updating client-go version to v0.18.6 in replace section as well https://github.com/openshift/machine-config-operator/blob/release-4.5/go.mod#L96 . go.mod replace section allows user to  pin a dependency to version projects want to stick over.

File: go.mod [hunk 1]
```diff
@@ -74,7 +74,7 @@ require (
 	k8s.io/api v0.18.0
 	k8s.io/apiextensions-apiserver v0.18.0
 	k8s.io/apimachinery v0.18.0
```
Comment (p0lyn0mial) [inline go.mod:76]: also, I think that you should update all deps to the same level 1.18.6

File: go.mod [hunk 2]
```diff
@@ -71,10 +71,10 @@ require (
 	golang.org/x/time v0.0.0-20190921001708-c4c64cad1fd0
 	google.golang.org/appengine v1.6.1 // indirect
 	gopkg.in/check.v1 v1.0.0-20190902080502-41f04d3bba15 // indirect
-	k8s.io/api v0.18.0
+	k8s.io/api v0.18.6
 	k8s.io/apiextensions-apiserver v0.18.0
```
Comment (p0lyn0mial) [inline go.mod:75]: I think that we should bring all k8s dependencies to the same level, `k8s.io/api/apiextensions-apiserver` as well.
Please check for others.

File: vendor/k8s.io/client-go/tools/cache/reflector.go [hunk 3]
```diff
@@ -256,13 +256,14 @@ func (r *Reflector) ListAndWatch(stopCh <-chan struct{}) error {
 			}
 
 			list, paginatedResult, err = pager.List(context.Background(), options)
-			if isExpiredError(err) {
-				r.setIsLastSyncResourceVersionExpired(true)
-				// Retry immediately if the resource version used to list is expired.
+			if isExpiredError(err) || isTooLargeResourceVersionError(err) {
```
Comment (p0lyn0mial) [inline vendor/k8s.io/client-go/tools/cache/reflector.go:259]: yes, this is the fix we want.

**Comment by sinnykumari** [comment]
LGTM
/approve

@p0lyn0mial Feel free to lgtm

File: go.mod [hunk 4]
```diff
@@ -88,26 +88,26 @@ replace (
...
+	k8s.io/component-base => k8s.io/component-base v0.18.6
+	k8s.io/cri-api => k8s.io/cri-api v0.18.6
+	k8s.io/csi-translation-lib => k8s.io/csi-translation-lib v0.18.6
+	k8s.io/kube-aggregator => k8s.io/kube-aggregator v0.18.6
+	k8s.io/kube-controller-manager => k8s.io/kube-controller-manager v0.18.6
+	k8s.io/kube-proxy => k8s.io/kube-proxy v0.18.6
+	k8s.io/kube-scheduler => k8s.io/kube-scheduler v0.18.6
+	k8s.io/kubectl => k8s.io/kubectl v0.18.6
+	k8s.io/kubelet => k8s.io/kubelet v0.18.6
 	k8s.io/kubernetes => k8s.io/kubernetes v1.18.0
```
Comment (p0lyn0mial) [inline go.mod:109]: you are not updating this to 1.18.6 ?

Comment (sinnykumari) [reply go.mod:109]: hah, these tiny little numbers ;)

Comment (kikisdeliveryservice) [reply go.mod]: thanks for your patience @p0lyn0mial Done.

**Comment by p0lyn0mial** [comment]
/lgtm

many thanks for the pr.

**Pull request merged**